strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiMetadataFileParser.java

```diff
@@ -0,0 +1,59 @@
+package org.carlspring.strongbox.util;
+
+import com.google.common.collect.Maps;
+import org.carlspring.strongbox.domain.PypiMetadataDao;
+import static org.carlspring.strongbox.domain.PypiMetadataDao.PypiMetadataKey;
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Map;
+
+@Component
+public class PypiMetadataFileParser {
+
+    public static PypiMetadataDao parseMetadataFile(File file) {
+        Map<String, String> keyValueMap = generateKeyValueMap(file);
+        PypiMetadataDao dao = new PypiMetadataDao();
+        return populateAnnotatedFields(dao, keyValueMap);
+    }
+
+    private static PypiMetadataDao populateAnnotatedFields(PypiMetadataDao object, Map<String, String> keyValueMap){
+        Class<?> clazz = object.getClass();
+        for (Field field: clazz.getDeclaredFields()) {
+            field.setAccessible(true);
+            try {
+                if (field.isAnnotationPresent(PypiMetadataKey.class)) {
+                    PypiMetadataKey pypiMetadataKey = field.getAnnotation(PypiMetadataKey.class);
+                    field.set(object, keyValueMap.get(pypiMetadataKey.name()));
+                }
+            } catch (IllegalAccessException ex) {
+                ex.printStackTrace();
+            }
+        }
+        return object;
+    }
+
+    private static Map<String, String> generateKeyValueMap(File file) {
+        Map<String, String> keyValueMap = Maps.newHashMap();
+
+        BufferedReader reader;
+        try {
+            reader = new BufferedReader(new FileReader(file));
+            String line = "";
+            while ((line = reader.readLine()) != null) {
+                String[] keysValues = line.split(":", 2);
+                keyValueMap.put(keysValues[0].trim(), keysValues[1].trim());
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Please, use loggers instead.

Reply:
Changed that. Please kindly check if I'm throwing exception right after logging it. Thanks!

Reply:
Well, no, you're not. :)
You're still printing the stacktrace.

Reply:
Thanks! :)